java/src/jmri/InstanceManagerAutoInitialize.java

```diff
@@ -0,0 +1,23 @@
+package jmri;
+
+/**
+ * Provide a hint to the {@link jmri.InstanceManager} that this object needs
+ * have additional initialization performed after it is made available by the
+ * InstanceManager. This allows two classes that have circular dependencies on
+ * being able to get the default instance of each other to be managed
+ * successfully.
+ * <p>
+ * Note: the need to have a class implement this probably is indicative of other
+ * design issues in the implementing class.
+ *
+ * @author Randall Wood Copyright 2017
+ */
+public interface InstanceManagerAutoInitialize {
```

Comment:
@pabender Agreed. I attached the PlantUML files that created them, happy to update those once I understand Randall's intent.  

In some cases, like maybe these, the diagram definition is big enough that it makes it hard to read the .java file.  (I still read the comments instead of going to the web for JavaDoc most of the time)  In those cases, I've manually made the files and linked them.  C.f. the files for the [ussctc package](https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/ussctc/package-info.java) in [java/src/jmri/jmrit/ussctc/doc-files](https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/ussctc/doc-files).